### DISCUSSION TITLE
Is it possible to change the font colors?

### DISCUSSION THREAD

**Discussion by nepomuk16321** [eyeonus/Trade-Dangerous#174] [Q&A] [answered]
Is it possible (easy) to adjust the colors? I have a dark background (black or dark blue). The dark fonts, for example, are very difficult to read on this background. I have already changed the background to a light green or ochre or similar, but then the light fonts are difficult to see again. It's mainly about the font in dark blue on a dark background.

**Comment by kfsone** [comment, marked as answer]
TD currently has two different mechanisms by which it adds color, can you show which commands and which text are a problem? I'm dark-mode all the way myself, and I use iterm3 when I on the mac and Windows Terminal on PC. If you are using the old Windows command window or powershell windows, your mileage will definitely vary and I strongly recommend you consider using Microsoft's replacement for it, Windows Terminal:
https://github.com/microsoft/terminal

If you're referring to the old coloring of the "trade run" command, that's definitely something we would like done but probably needs a volunteer to tackle if you want it done sooner rather than later.

For anyone who reads this and wonders what is involved:

TD now uses Python's [rich](https://rich.readthedocs.io/en/stable/introduction.html) library to perform console styling, including the new automatic coloring of things like links, the progress bars, etc. When you print via the rich methods - and the provide a drop-in `print` replacement - you can do bbcode-like things: `:information_source: [red]ERROR[/]` (prints ℹ️ followed by ERROR in red).

There are two fundamental components of every TD operation: The "environment" (TradeEnv) and the database (TradeDB), and here TradeEnv (usually `tdenv`) is key:

Its role is to know what the user/program is currently trying to do, and how - so it knows all the command line settings, if we add a config file, it will know the config, etc.

So it also provides methods for output, and all td output is supposed to go through it.

When I introduced the rich code, I also gave TradeEnv a concept of [themes](https://github.com/eyeonus/Trade-Dangerous/blob/release/v1/tradedangerous/tradeenv.py#L38-L99), but at the moment there's no way to select themes and there's only three themes (Base, BasicRich, RichColor).

AFAIK the only code using it is TradeEnv itself and possibly a couple of pieces of the eddblink plugin. See
https://github.com/eyeonus/Trade-Dangerous/blob/release/v1/tradedangerous/tradeenv.py#L137

**Comment by kfsone** [comment]
Actually, I stand corrected, you *can* switch between the BasicRichTheme and the RichColorTheme - it turns on the additional coloration when you use --color, but the only thing that will currently affect is certain output lines generated by TradeEnv itself.

**Reply by nepomuk16321** [reply]
Hello kfsone, thank you very much for the explanation. It's all a bit too much for me. I'm missing the background. I had hoped to perhaps change an assignment in a file or something similar. I have now adjusted the background of the PowerShell again. It's OK for me now. Thanks again !
